foundation/blockchain/state.go

```diff
@@ -513,6 +513,7 @@ func (s *State) MineNewBlock(ctx context.Context) (Block, time.Duration, error)
 		// Apply the balance changes based on this transaction. Set status
 		// information for other nodes to process this correctly.
 		if err := applyTransactionToBalance(balanceSheet, tx); err != nil {
+			// why do we still mine transaction if we have error in it?
```

Comment:
Because they need to be removed in the mempool of the other minors. There are transactions that fail in blocks on Ethereum.